### PR TITLE
Auth rework, some clean up

### LINKED ISSUE #95: User can log out
**User story:** As a user, I want to be able to log out of the application so I don't leave access to sensitive data when I step away from my computer.

**AC:**
* I can log out by click "Log out" in the Header
* Access and refresh tokens are removed from Client
* Access and refresh tokens are invalidated at the authentication server

### LINKED ISSUE #217: Add persistent access token
Currently, the access token is only stored in React's internal state. This prevents creating a new tab without signing in again or a longer-than-browser-session authentication.

**Story:** As a user, I want to be able to sign in to Hedwig and not be prompted to sign in again until I a) sign out, b) close all browser windows, c) or 1 hour; whichever occurs first.

**AC:**
- The access and refresh token are persisted as a cookie or to local or session storage.
- If an expired token is stored, it prompts the user to sign in again.
- Logging out (which doesn't exist yet [see #95]) clears all stored tokens.

### PR DESCRIPTION
**Pull request by cailyncodes**
* Rename LoginContext to AuthenticationContext
* Add comments
* Make accessToken persistent
* Support soft refresh logout
* Remove some react warnings
* State-ify more of AuthenticationContext

Should close #95; close #217

### CODE REVIEW

File: src/Hedwig/ClientApp/src/index.tsx [hunk 1]
```diff
@@ -7,18 +7,19 @@ import { BrowserRouter } from 'react-router-dom';
 import ApolloClient from 'apollo-boost';
 import { ApolloProvider } from '@apollo/react-hooks';
 import * as serviceWorker from './serviceWorker';
-import { LoginProvider } from './contexts/Login';
+// import { LoginProvider } from './contexts/Login/LoginContext';
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/index.tsx]: leftover commented out code

File: src/Hedwig/ClientApp/src/containers/Enrollment/_sections/EnrollmentFunding.tsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ const EnrollmentFunding: Section = {
   },
 
   Form: ({ enrollment, mutate, callback }) => {
-
+    console.log("test");
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/containers/Enrollment/_sections/EnrollmentFunding.tsx]: Oops

File: src/Hedwig/ClientApp/src/containers/Enrollment/_sections/FamilyIncome.tsx [hunk 3]
```diff
@@ -32,6 +32,7 @@ const FamilyIncome: Section = {
 	},
 
 	Form: ({ enrollment, mutate, callback }) => {
+		console.log("family income", enrollment);
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/containers/Enrollment/_sections/FamilyIncome.tsx]: Oops too

File: src/Hedwig/ClientApp/src/containers/Enrollment/_sections/EnrollmentFunding.tsx (hunk 2 above)
Comment (jlhogan) [reply src/Hedwig/ClientApp/src/containers/Enrollment/_sections/EnrollmentFunding.tsx]: somehow i noticed the dangling comments, but not the new console logs ? hahah

**Pull request merged**